src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/BlazorDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,87 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+
+import { RazorLogger } from '../RazorLogger';
+import { onDidTerminateDebugSession } from './TerminateDebugHandler';
+
+export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+
+    constructor(private readonly logger: RazorLogger, private readonly vscodeType: typeof vscode) { }
+
+    public async resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, configuration: vscode.DebugConfiguration): Promise<vscode.DebugConfiguration | undefined> {
+        const shellPath = process.platform === 'win32' ? 'cmd.exe' : 'dotnet';
+        const shellArgs = process.platform === 'win32' ? ['/c', 'chcp 65001 >NUL & dotnet run'] : ['run'];
+        const spawnOptions = {
+            cwd: configuration.cwd || (folder && folder.uri && folder.uri.fsPath),
+        };
+
+        const output = this.vscodeType.window.createTerminal({
+          name: 'Blazor WebAssembly App',
+          shellPath,
+          shellArgs,
+          ...spawnOptions,
+        });
+
+        /**
+         * On non-Windows platforms, we need to terminate the Blazor
+         * dev server and its child processes.
+         */
+        const terminate = this.vscodeType.debug.onDidTerminateDebugSession(async event => {

```

Comment:
This will fire when any debug session finishes -- you want to look at the session to make sure it's the one we care about. (e.g. I might have an API server written in some other language, stopping/restarting that shouldn't cause my blazor debugging to stop)

Reply:
Seems sensible. The `type` property on the DebugSession seemed like the best thing to check against but let me know if there is a better approach here.